tests/upgrade/0.8.2-0.9.0/jiva/volume_test.go

```diff
@@ -72,6 +78,39 @@ var _ = Describe("[jiva] TEST VOLUME UPGRADE", func() {
 			podCount := ops.GetPodRunningCountEventually("default", upgradedLabel, replicaCount+1)
 			Expect(podCount).To(Equal(replicaCount+1), "while checking pod version")
 
+			By("verifying registered replica count and replication factor")
+			podList, err := ops.PodClient.List(metav1.ListOptions{LabelSelector: ctrlLabel})
+			Expect(err).ShouldNot(HaveOccurred(), "while listing controller pod")
+
+			replicationFactor := ""
+			for _, env := range podList.Items[0].Spec.Containers[0].Env {
+				if env.Name == "REPLICATION_FACTOR" {
+					replicationFactor = env.Value
+				}
+			}
+			Expect(replicationFactor).ToNot(Equal(""), "while fetching replication factor")
+
+			curl := "curl http://localhost:9501/v1/volumes"
+			cmd := []string{"/bin/bash", "-c", curl}
+			opts := tests.NewOptions().
+				WithPodName(podList.Items[0].Name).
+				WithNamespace(podList.Items[0].Namespace).
+				WithContainer(podList.Items[0].Spec.Containers[0].Name).
+				WithCommand(cmd...)
+
+			out, err := ops.ExecPod(opts)
+			Expect(err).To(BeNil(), "while executing command in container ", cmd)
+
+			volumes := jiva.VolumeCollection{}
+			err = json.Unmarshal(out, &volumes)
+			Expect(err).To(BeNil(), "while unmarshalling volumes", string(out))
+
+			registeredReplicaCount := strconv.Itoa(volumes.Data[0].ReplicaCount)
```

Comment:
@shubham14bajpai Do you think we should have `pkg/ops/jiva/v1alpha1/jiva.go` that provides us with the registered replica count? It will use pod exec internally.

This is obviously a future thing.
I have tried to put some initial thoughts here:
https://github.com/AmitKumarDas/Decisions/blob/master/blogs/oep/0001_maya_operations.md

Reply:
That could be good thing as this check will be used more often.